incubator/hnc/test/e2e/acting-as.go

```diff
@@ -0,0 +1,42 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "sigs.k8s.io/multi-tenancy/incubator/hnc/pkg/testutils"
+)
+
+var _ = Describe("Acting-as", func() {
+	const (
+		parent = "parent"
+		child  = "child"
+	)
+
+	BeforeEach(func() {
+		CleanupNamespaces(parent, child)
+	})
+
+	AfterEach(func() {
+		CleanupNamespaces(parent, child)
+	})
+
+	It("should allow acting as different service accounts", func() {
+		// create the namespaces
+		MustRun("kubectl create ns", parent)
+		MustRun("kubectl create ns", child)
+
+		// fail to set parent
+		MustNotRun("kubectl hns set", child, "--parent", parent, "--as system:serviceaccount:"+child+":default")
+
+		// set the proper roles giving the required permissions
+		MustRun("kubectl -n", child, "create role "+child+"-hnc-role --verb=get,create,update --resource=HierarchyConfiguration")
+		MustRun("kubectl -n", child, "create rolebinding "+child+"-hnc-rolebinding --role "+child+"-hnc-admin --serviceaccount="+child+":default")
```

Comment:
Nit: instead of string manipulation, you can just use commas, which may be more natural, e.g.:

```
MustRun("kubectl -n", child, "create rolebinding", child+"-hnc-rolebinding", "--role", child+"-hnc-admin", "--serviceaccount", child+":default")
```

or if you're using the `admin` clusterrole:

```
MustRun("kubectl -n", child, "create rolebinding local-admin --clusterrole admin --serviceaccount", child+":default")
```

Note that the rolebinding name doesn't need to include `child` in the name; "local-admin" would be fine.